### PR TITLE
M3 IMPL - PR number 1

### PR DESCRIPTION
**Pull request by Sidarthapati**
# Pull Request

## Summary

M3 code addition: This PR adds basic flyway migration scripts for 9 tables, all a part of M3 module. Also adds a base entity, every other dto would extend this. THis PR also has several serviceability files with mock data for testing purposes.

---

## What Changed

-  Flyway migration scripts added (v1-v9)
-  Serviceability mock data added
-  Base entity created (would be extended by all other jpa entities and dtos)

---

## Why This Change?

There was no implementation of M#. This modules add the basic prerequisites for M3, namely tables and mock data.

---

## Testing
- [x] Tested locally
- [x] Edge cases considered
- [x] No breaking changes introduced

### Test Evidence

     
     
Screenshot 2026-05-13 at 1 55 23 AM

---

## Impact
No impact. Base code added for M3 , no regression, no rollback.

### Areas Affected

- [ ] Backend
- [ ] Frontend
- [ ] Routing
- [ ] Infra
- [x] Database
- [ ] NA

### Modules Affected

- [ ] M1 — Auth
- [ ] M2 — Pricing
- [x] M3 — Grid
- [ ] M4 — Orders
- [ ] M5 — Dispatch
- [ ] M6 — Routing
- [ ] M7 — Hub
- [ ] M8 — Barcode
- [ ] M9 — Airline
- [ ] M10 — SLA
- [ ] M11 — Exceptions
- [ ] NA

---

## Risks / Concerns

- None: Addition of base code with no regression scope. If required can always remove files.

---

## Checklist
- [x] Code follows project conventions (package layout, no cross-module internal imports)
- [x] Self-review completed
- [x] Append-only audit trail preserved (no mutations to scan/manifest/role-change records)
- [x] No sensitive data or credentials exposed
- [x] Documentation updated if behaviour changed
- [x] Ready for review

### CODE REVIEW

File: grid/src/main/resources/db/migration/V2__create_tile.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+CREATE TABLE tile (
+    id                UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    grid_id           UUID NOT NULL REFERENCES grid(id),
+    row_idx           INT NOT NULL,
+    col_idx           INT NOT NULL,
+    is_active         BOOLEAN NOT NULL DEFAULT false,
+    traversal_cap_sec INT,
```
Comment (BanerjeeAgniva) [inline grid/src/main/resources/db/migration/V2__create_tile.sql:7]: What does this mean?

File: grid/src/main/resources/db/migration/V6__create_tile_demand_snapshot.sql [hunk 2]
```diff
@@ -0,0 +1,14 @@
+CREATE TABLE tile_demand_snapshot (
+    id                    UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    tile_id               UUID NOT NULL REFERENCES tile(id),
+    snapshot_date         DATE NOT NULL,
+    hist_avg_orders       DOUBLE PRECISION NOT NULL,
+    current_orders        INT NOT NULL,
+    demand_score_orders   DOUBLE PRECISION NOT NULL,
```
Comment (BanerjeeAgniva) [inline grid/src/main/resources/db/migration/V6__create_tile_demand_snapshot.sql:7]: what does this mean?

**Review by BanerjeeAgniva — APPROVED** [review]

File: grid/src/main/resources/db/migration/V2__create_tile.sql (hunk 1 above)
Comment (Sidarthapati) [reply grid/src/main/resources/db/migration/V2__create_tile.sql:7]: This is the OSRM road-travel time in seconds from a tile's south-west corner to its north-east corner — i.e., the longest road trip possible within a single 2 km × 2 km tile.

  Its sole purpose is as a winsorisation ceiling for inter-stop travel measurement. When the nightly replan computes how long a DA spends travelling between consecutive pickups within a tile, a single
  anomalous trip (traffic incident, wrong turn) can inflate the average dramatically. Before averaging, we clamp each individual measurement at traversal_cap_sec: if a pair took longer than a full diagonal
  crossing of the tile, something non-routine happened and that data point shouldn't skew the demand estimate.
  
  For a 2 km × 2 km tile at Indian city speeds, this is typically 360–600 seconds (6–10 minutes). It's NULL until the OSRM matrix refresh runs; the nightly job skips winsorisation for tiles with a null cap
  and falls back to the city-wide average instead.

File: grid/src/main/resources/db/migration/V6__create_tile_demand_snapshot.sql (hunk 2 above)
Comment (Sidarthapati) [reply grid/src/main/resources/db/migration/V6__create_tile_demand_snapshot.sql:7]: This is the blended order-count forecast for a tile: 0.70 × hist_avg_orders + 0.30 × current_orders.

  The 70/30 split is the platform's demand weighting model — 70% historical (7-day rolling average) for stability, 30% current day for reactivity. hist_avg_orders and current_orders are the two raw inputs;
  demand_score_orders is their weighted combination.

  Critically, this is an intermediate value — it is not what the CP-SAT solver consumes. The solver uses demand_score_minutes (= demand_score_orders × order_engaged_min_per_order), which converts order count
  into DA shift-time. demand_score_orders is retained in the snapshot for station manager reporting and audit — so the station manager can see "this tile is forecast at 14.2 orders" alongside the time-based
  utilisation estimate.
